WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -760,22 +766,17 @@ private extension ProductFormViewController {
 //
 private extension ProductFormViewController {
 
-    func updateNavigationBar(isUpdateEnabled: Bool) {
-        var rightBarButtonItems = [UIBarButtonItem]()
-
-        switch viewModel.formType {
-        case .add:
-            rightBarButtonItems.append(createPublishBarButtonItem())
-        case .edit:
-            if isUpdateEnabled {
-                rightBarButtonItems.append(createUpdateBarButtonItem())
+    func updateNavigationBar() {
+        // Create action buttons based on view model
+        let rightBarButtonItems: [UIBarButtonItem] = viewModel.actionButtons.reversed().compactMap { buttonType in
+            switch buttonType {
+            case .publish:
+                return createPublishBarButtonItem()
+            case .save:
+                return createSaveBarButtonItem()
+            case .more:
+                return createMoreOptionsBarButtonItem()
```

Comment:
👏  This feels really clear and easy to follow, especially with the conditions all moved to the view model.